### PR TITLE
Fix/geolocation number

### PR DESCRIPTION
**Pull request by kaisermann**
#### What is the purpose of this pull request?

Fix geolocation flow when the address query doesn't contain the address number. Currently, it throws an error because it's passing the `SyntheticEvent` instead of the input `value`. Also, the address form is trying to validate the (hidden) `postalCode` field which can be empty if no number is defined on the `addressQuery`.

#### What problem is this solving?

https://app.clubhouse.io/vtex/story/26043/my-account-address-campo-de-n%C3%BAmero-exibindo-referencia-do-object-para-lojas-do-uruguay

#### How should this be manually tested?

1) https://kiwi--recorrenciaqa.myvtex.com/account#/addresses
2) `New address`
3) Type `av`, select an address without a number
4) Type the number on the specified field.
5) Submit

#### Screenshots or example usage

**Before**:
image

**After**:
image

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Fizxed
```
Comment (klzns) [inline CHANGELOG.md]:
```suggestion
### Fixed
```

File: react/inputs/StyleguideInput/index.js [hunk 2]
```diff
@@ -164,18 +164,18 @@ class StyleguideInput extends Component {
...
               ref={inputRef}
-              value={field.value}
+              value={address[field.name].value || ''}
             />
           </div>
           <div className="vtex-address-form__number-checkbox flex flex-row ml7 mt6 w-50">
             <Checkbox
               id="option-0"
-              label="Option 0"
+              label={this.props.notApplicableLabel || 'N/A'}
```
Comment (salesfelipe) [inline react/inputs/StyleguideInput/index.js]: Maybe you should internationalize the 'N/A' like: https://github.com/vtex-apps/my-account/blob/330f4d6ca43d6bc0074fe002024034b7fdffa42b/messages/pt.json#L44

File: react/PostalCodeGetter.test.js [hunk 3]
```diff
@@ -7,14 +7,14 @@ import useOneLevel from './country/__mocks__/useOneLevel'
...
-describe.only('PostalCodeGetter', () => {
+describe('PostalCodeGetter', () => {
   it('renders without crashing', () => {
     render(
       <PostalCodeGetter
         address={address}
         rules={usePostalCode}
         onChangeAddress={jest.fn()}
-      />
+      />,
```
Comment (kevinch) [inline react/PostalCodeGetter.test.js]: Comma added by lint?

File: react/AddressRules.js [hunk 4]
```diff
@@ -18,57 +20,82 @@ class AddressRules extends Component {
...
-        }
-        this.setState({ rules: defaultRules })
-        return defaultRules
-      }
-      if (process.env.NODE_ENV !== 'production') {
-        console.warn('An unknown error occurred.')
+    let rules = await this.fetchRules(rulePromise)
+
+    if (useGeolocation && rules.geolocation) {
+      rules = {
```
Comment (salesfelipe) [inline react/AddressRules.js]: Instead of destructuring, why don't you just assign like:

rules["_usingGeolocationRules"] = true ...

Comment (kaisermann) [reply react/AddressRules.js]: I don't want to modify the original object (I dunno if it's cached and another `import(...)` would return the same modified object)

File: react/PostalCodeGetter.test.js (hunk 3 above)
Comment (kaisermann) [reply react/PostalCodeGetter.test.js]: Yeah, but it shouldn't add trailing comma for function args. Will fix this

File: react/AddressRules.js [hunk 5]
```diff
@@ -18,57 +20,82 @@ class AddressRules extends Component {
...
-    if (!result) return false
-    return result[1]
+    this.setState({ rules })
+    return rules
   }
 
   render() {
     const { children } = this.props
-    const { rules } = this.state
+    let { rules } = this.state
```
Comment (klzns) [inline react/AddressRules.js]: Why not const?

File: react/inputs/StyleguideInput/index.js [hunk 6]
```diff
@@ -158,24 +157,18 @@ class StyleguideInput extends Component {
...
-                }),
-              })}
-              onChange={this.props.onChange}
+              onChange={this.handleChange}
               onBlur={this.props.onBlur}
               disabled={loading || disabled}
               error={!this.state.isInputValid}
               ref={inputRef}
-              value={field.value}
+              value={address[field.name].value || ''}
```
Comment (klzns) [inline react/inputs/StyleguideInput/index.js]: Isn't `field.value` the same value?

Comment (kaisermann) [reply react/inputs/StyleguideInput/index.js]: @klzns Nope, `field` is pointing to the `rules.fields[...]`. It contains all the field rules, but the value lives in the `address[fieldName].value` prop

File: react/AddressRules.js (hunk 5 above)
Comment (kaisermann) [reply react/AddressRules.js]: Ma bad!

File: react/inputs/StyleguideInput/index.js (hunk 6 above)
Comment (klzns) [reply react/inputs/StyleguideInput/index.js]: oh, you're right

**Pull request merged**